lib/install/vchlog/buffered_pipe.go

```diff
@@ -37,24 +41,28 @@ func NewBufferedPipe() *BufferedPipe {
 	var m sync.Mutex
 	c := sync.NewCond(&m)
 	return &BufferedPipe{
-		buffer: bytes.NewBuffer(nil),
-		c:      c,
-		closed: false,
+		buffer:      bytes.NewBuffer(nil),
+		c:           c,
+		readClosed:  false,
+		writeClosed: false,
+		readerReady: false,
 	}
 }
 
 // Read is blocked until a writer in the queue is done writing (until data is available)
 func (bp *BufferedPipe) Read(data []byte) (n int, err error) {
 	bp.c.L.Lock()
 	defer bp.c.L.Unlock()
+	defer bp.c.Broadcast()
 
-	// pipe closed, drop all left-over data
-	if bp.closed {
-		return 0, io.EOF
-	}
-	for bp.buffer.Len() == 0 && !bp.closed {
+	bp.readerReady = true // the reader is ready to read
```

Comment:
I'd update the comment to something like: `// We have a valid consumer so switch to flushing in Close`